### PR TITLE
feat(CO-19): Frontend equipped inventory panel

### PR DESCRIPTION
**Pull request by lakospal0**

### CODE REVIEW

File: src/app/services/api/api.service.ts [hunk 1]
```diff
@@ -15,4 +16,5 @@ export class ApiService {
       title: '¡Hola Jorge!'
     });
   }
+
```
Comment (dyanimay) [inline src/app/services/api/api.service.ts]: ánnesszeszeri lájn

File: src/app/app-routing.module.ts [hunk 2]
```diff
@@ -11,6 +11,7 @@ import { LoginPageComponent } from './components/login-page/login-page.component
 import { InventoryPageComponent } from './components/inventory-page/inventory-page.component';
 import { DungeonComponent } from './components/dungeon/dungeon.component';
 
+
```
Comment (dyanimay) [inline src/app/app-routing.module.ts]: ánnesszeszeri, 1 empty line is enough between the last import and "const"

File: src/app/components/single-hero-page/single-hero-page.component.ts [hunk 3]
```diff
@@ -2,22 +2,31 @@ import { HeroModel } from './../../models/heroModel';
...
 import { HeroService } from 'src/app/services/heroservice/hero.service';
+import { Equipment } from 'src/app/models/equipment';
 
 @Component({
   selector: 'app-single-hero-page',
   templateUrl: './single-hero-page.component.html',
   styleUrls: ['./single-hero-page.component.scss']
 })
 export class SingleHeroPageComponent implements OnInit {
+
```
Comment (dyanimay) [inline src/app/components/single-hero-page/single-hero-page.component.ts]: ánnesszeszeri lájn

File: src/app/models/equipment.ts [hunk 4]
```diff
@@ -1,12 +1,14 @@
...
 
 export class Equipment {
+  private _id: number;
   private _name: string;
   private _modifiers: Array<AttributeModifier>;
   private _type: string;
   private _active: boolean;
 
-  constructor(name: string, modifier: Array<AttributeModifier>, type: string, active: boolean) {
+  constructor(id: number, name: string, modifier: Array<AttributeModifier>, type: string, active: boolean) {
```
Comment (juhaszPeterDev) [inline src/app/models/equipment.ts]: refactor to a constructor with one param

File: src/app/models/heroModel.ts [hunk 5]
```diff
@@ -172,4 +174,10 @@ export class HeroModel {
     }));
     return this._defense + defenseModifiers;
   }
+
+  public activeEquipments(): Equipment[] {
+    return this.inventory.filter((equipment: Equipment) => equipment.active === true).map((element) => {
```
Comment (juhaszPeterDev) [inline src/app/models/heroModel.ts]: this is equal to "filter((equipment: Equipment) => equipment.active)"

File: src/app/services/api/api.service.ts [hunk 6]
```diff
@@ -2,6 +2,7 @@ import { Injectable } from '@angular/core';
 import { Observable, of } from 'rxjs';
 
 import { HelloWorld } from '../../models/hello-world';
+import { Equipment } from '../../models/equipment';
```
Comment (juhaszPeterDev) [inline src/app/services/api/api.service.ts]: remove this

File: src/app/components/equipped-inventory/equipped-inventory.component.ts [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+    waist: 'Waist',
+    legs: 'Legs'
+  };
+
+  constructor() { }
+
+  ngOnInit() {
+    const activeItems = this.hero.activeEquipments();
+
+    /* Loop trough the types to show */
```
Comment (matyaspeterfi) [inline src/app/components/equipped-inventory/equipped-inventory.component.ts]: Comment left in

File: src/app/components/equipped-inventory/equipped-inventory.component.ts [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+    for (const key in this.typeToLabel) {
+      if (this.typeToLabel.hasOwnProperty(key)) {
+        const typeToLookUp = key;
+
+        /* Loop trough the active items, to see if we can find the typeToLookUp*/
```
Comment (matyaspeterfi) [inline src/app/components/equipped-inventory/equipped-inventory.component.ts]: Another comment

File: src/app/components/equipped-inventory/equipped-inventory.component.ts [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+  ngOnInit() {
+    const activeItems = this.hero.activeEquipments();
+
+    for (const key in this.typeToLabel) {
+      if (this.typeToLabel.hasOwnProperty(key)) {
+        const typeToLookUp = key;
+        let typeFound = false;
+        for (const item in activeItems) {
+          if (activeItems.hasOwnProperty(item)) {
+            if (activeItems[item].type === typeToLookUp) {
```
Comment (juhaszPeterDev) [inline src/app/components/equipped-inventory/equipped-inventory.component.ts]: image

**Pull request merged**